builder/llpcBuilderImplSubgroup.cpp

```diff
@@ -0,0 +1,1088 @@
+    }
+    else
+    {
+        Value* pCompare = CreateICmpEQ(pIndex, getIntN(indexBits, 0));
+        pResult = CreateSelect(pCompare, CreateDsSwizzle(pIndex, GetDsSwizzleQuadMode(0, 0, 0, 0)), pResult);
+
+        pCompare = CreateICmpEQ(pIndex, getIntN(indexBits, 1));
+        pResult = CreateSelect(pCompare, CreateDsSwizzle(pIndex, GetDsSwizzleQuadMode(1, 1, 1, 1)), pResult);
+
+        pCompare = CreateICmpEQ(pIndex, getIntN(indexBits, 2));
+        pResult = CreateSelect(pCompare, CreateDsSwizzle(pIndex, GetDsSwizzleQuadMode(2, 2, 2, 2)), pResult);
+
+        pCompare = CreateICmpEQ(pIndex, getIntN(indexBits, 3));
+        pResult = CreateSelect(pCompare, CreateDsSwizzle(pIndex, GetDsSwizzleQuadMode(3, 3, 3, 3)), pResult);
+    }
+
+    return pResult;
+}
+
+// =====================================================================================================================
+// Create a subgroup quad swap horizontal call.
+Value* BuilderImplSubgroup::CreateSubgroupQuadSwapHorizontal(
+    Value* const pValue,   // [in] The value to swap.
+    const Twine& instName) // [in] Name to give final instruction.
+{
+    if (SupportDpp())
+    {
+        return CreateDppMov(pValue, DppCtrl::DppQuadPerm1032, 0xF, 0xF, false);
+    }
+    else
+    {
+        return CreateDsSwizzle(pValue, GetDsSwizzleQuadMode(2, 3, 0, 1));
+    }
+}
+
+// =====================================================================================================================
+// Create a subgroup quad swap vertical call.
+Value* BuilderImplSubgroup::CreateSubgroupQuadSwapVertical(
+    Value* const pValue,   // [in] The value to swap.
+    const Twine& instName) // [in] Name to give final instruction.
+{
+    if (SupportDpp())
+    {
+        return CreateDppMov(pValue, DppCtrl::DppQuadPerm2301, 0xF, 0xF, false);
+    }
+    else
+    {
+        return CreateDsSwizzle(pValue, GetDsSwizzleQuadMode(1, 0, 3, 2));
+    }
+}
+
+// =====================================================================================================================
+// Create a subgroup quadswapdiagonal call.
+Value* BuilderImplSubgroup::CreateSubgroupQuadSwapDiagonal(
+    Value* const pValue,   // [in] The value to swap.
+    const Twine& instName) // [in] Name to give final instruction.
+{
+    if (SupportDpp())
+    {
+        return CreateDppMov(pValue, DppCtrl::DppQuadPerm0123, 0xF, 0xF, false);
+    }
+    else
+    {
+        return CreateDsSwizzle(pValue, GetDsSwizzleQuadMode(3, 2, 1, 0));
+    }
+}
+
+// =====================================================================================================================
+// Create The group arithmetic operation identity.
+Value* BuilderImplSubgroup::CreateGroupArithmeticIdentity(
+    GroupArithOp   groupArithOp, // The group arithmetic operation to get the identity for.
+    Type* const    pType)        // [in] The type of the identity.
+{
+    switch (groupArithOp)
+    {
+        case GroupArithOp::IAdd:
```

Comment:
The indentation for case branch is error

switch (...)
{
case XXX:
case YYY:
}